src/mainwindow.cpp

```diff
@@ -697,6 +700,37 @@ void MainWindow::find()
     consoleTabulator->terminalHolder()->currentTerminal()->impl()->toggleShowSearchBar();
 }
 
+void MainWindow::sendHistoryToEditor()
+{
+    const QString fn = QStandardPaths::writableLocation(QStandardPaths::TempLocation) + QLatin1String("/qterminal.history.") + QString::number(QCoreApplication::applicationPid());
```

Comment:
The first issue - /tmp is often world-readable, while terminal outputs may contain sensitive information. I'd suggest not to save the history here.